pizzalab/pizzalabapp/tests/test_serializers.py

```diff
@@ -0,0 +1,92 @@
+import io
+
+from rest_framework.parsers import JSONParser
+from rest_framework.request import Request
+from rest_framework.reverse import reverse
+from rest_framework.test import APIRequestFactory, APITestCase
+
+from pizzalabapp.models import OrderItem, Pizza
+from pizzalabapp.serializers import OrderItemSerializer, PizzaSerializer
+
+from .helpers import setup_pizzas, setup_return_order
+
+
+class PizzaTestCase(APITestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        setup_pizzas(1, True, False)
+
+    def setUp(self):
+        self.pizza = Pizza.objects.get(name="Pizza 0")
+
+        # HyperlinkedIdentityField requiring Request workaround:
+        factory = APIRequestFactory()
+        request = factory.get('/')
+        self.serializer_context = {
+            'request': Request(request),
+        }
+
+        self.pizza_serializer = PizzaSerializer(
+            instance=self.pizza,
+            context=self.serializer_context
+        )
+
+    def test_inexistent_ingredients_invalid(self):
+        self.pizza_serializer.data['ingredients'] += [{"name": "NewIngrr"}]
+        new_pizza_data = self.pizza_serializer.data
+        new_pizza_serializer = PizzaSerializer(data=new_pizza_data)
+        self.assertFalse(new_pizza_serializer.is_valid())
+        self.assertIsNotNone(new_pizza_serializer.errors.get('ingredients'))
+
+    def test_edit_price(self):
+        # get existing pizza details
+        current_content = self.client.get(
+            reverse('pizza-detail', args=[self.pizza.id])
+        ).content
+
+        # setup data with updated price
+        stream = io.BytesIO(current_content)
+        data = JSONParser().parse(stream)
+        data['price'] = 13
+        new_data = data
```

Comment:
You should reverse these if you want to have data != new_data:
new_data = data
new_data['price'] = 13
Otherwise you can as well use data in the following statements